edk2toollib/uefi/edk2/parsers/dsc_parser.py

```diff
@@ -237,16 +238,35 @@ def ParseInfPathMod(self, line):
     def __ProcessMore(self, lines, file_name=None):
         if(len(lines) > 0):
             for index in range(0, len(lines)):
-                (line, add, new_file) = self.__ParseLine(lines[index], file_name=file_name, lineno=index + 1)
-                if(len(line) > 0):
-                    self.Lines.append(line)
-                self.__ProcessMore(add, file_name=new_file)
+                try:
+                    (line, add, new_file) = self.__ParseLine(lines[index], file_name=file_name, lineno=index + 1)
+                    if(len(line) > 0):
+                        self.Lines.append(line)
+                    self.__ProcessMore(add, file_name=new_file)
+                except Exception as e:
+                    if not self._no_fail_mode:
+                        raise
+                    else:
+                        self.Logger.warning(e)
 
     def __ProcessDefines(self, lines):
         if(len(lines) > 0):
             for l in lines:
-                (line, add) = self.__ParseDefineLine(l)
-                self.__ProcessDefines(add)
+                try:
```

Comment:
why is this needed?

Reply:
Because we do include processing here. I was trying to catch the lines where we processed includes

Reply:
why do we do it in two spots?

Reply:
Because the DSC parser is a two pass system

Reply:
It processes it once to find defines (including in includes), then does the proper parse. This is still a hold-over from the limited parser that was pre project mu